### PR TITLE
filter out personal sites

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Adds a server-side filter that removes all
personal (aka: oneDrive) sites from the drive
retrieval.

The approach is not my preferred method.  I'd rather
have used a filter in the query parameters.  In fact, I
have a functioning query in postman:
`$filter=NOT contains(webUrl, 'sharepoint.com/personal/')`
But I can't seem to get the graph client query to build
when I add that to the query params.

## Type of change

- [x] :sunflower: Feature

## Issue(s)

* #1506

## Test Plan

- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/graph_connector.go [hunk 1]
```diff
@@ -220,6 +220,8 @@ func (gc *GraphConnector) setTenantSites(ctx context.Context) error {
 
 var errKnownSkippableCase = errors.New("case is known and skippable")
 
+const personalSitePath = "sharepoint.com/personal/"
```
Comment (ashmrtn) [inline src/internal/connector/graph_connector.go:223]: this is a bit more of a long-term question, but do we have any way to know if this is a locale independent way to specify personal sites?

File: src/internal/connector/graph_connector_test.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func (suite *GraphConnectorIntegrationSuite) TestSetTenantUsers() {
 
 	newConnector.graphService = *service
 
-	suite.Equal(len(newConnector.Users), 0)
+	suite.Equal(0, len(newConnector.Users))
```
Comment (ashmrtn) [inline src/internal/connector/graph_connector_test.go:69]: nit
```suggestion
	suite.Empty(newConnector.Users)
```

File: src/internal/connector/graph_connector.go [hunk 3]
```diff
@@ -237,6 +239,12 @@ func identifySite(item any) (string, string, error) {
 		return "", "", errors.Errorf("no name for Site: %s", *m.GetId())
 	}
 
+	// personal (ie: oneDrive) sites have to be filtered out server-side.
+	url := m.GetWebUrl()
+	if url != nil && strings.Contains(*url, personalSitePath) {
```
Comment (ashmrtn) [inline src/internal/connector/graph_connector.go:244]: I'm not familiar with how sharepoint site URLs are formatted, but is it possible for users to generate a site with a URL that would have `sharepoint.com/personal` in it but not be a personal site?

**Review by ashmrtn — APPROVED** [review]

File: src/internal/connector/graph_connector.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/connector/graph_connector.go:223]: Good question. Unfortunately not; the [site](https://learn.microsoft.com/en-us/graph/api/resources/site?view=graph-rest-1.0) sttruct is pretty sparse.  The next closest data point we have is `siteCollection.hostname`, where personal sites all have the domain `<tenant>-my.sharepoint.com`.  But, then again, that could be subject to locale as well.

File: src/internal/connector/graph_connector.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/internal/connector/graph_connector.go:244]: Looks like the slashes get stripped from the generated url.

image

File: src/internal/connector/graph_connector_test.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/connector/graph_connector_test.go:69]: All this well curated fanciness to keep track of.

**Pull request merged**